feat(clp-s): Add support for querying kv-pairs in the archive range index. (#929)
y-scope/clp: docs/src/user-guide/reference-json-search-syntax.md

\@key: value
```

### Querying kv-pairs in the archive range index

The archive range index maps metadata about each file ingested into an archive to a logical range
of the archive. Filters that reference fields in this metadata transparently get converted to
filters that match corresponding logical ranges of an archive.

To search for records corresponding to a kv-pair within the archive range index you can prefix the

> key with `$`. For example, the following query searches for records from a file that was passed to

Reviewer: should we mention the range index fields we currently support?

Author: I'll mention them in the SFA format design doc that I'm writing, but I think the current example with `$_filename` should be enough for most users.

The `$_archive_creator_id` and `$_file_split_number` fields will probably only be needed by ourselves and advanced users.

Reviewer: Got it, that makes sense

> ### Querying kv-pairs in the archive range index
> 
> The archive range index maps metadata about each file ingested into an archive to a logical range
> of the archive. Filters that reference fields in this metadata transparently get converted to
> filters that match corresponding logical ranges of an archive.
> 
> To search for records corresponding to a kv-pair within the archive range index you can prefix the
> key with `$`. For example, the following query searches for records from a file that was passed to
> compression as `test.jsonl`:
> 
> ```
> $_filename: "test.jsonl"
> ```

Reviewer 2: ````suggestion
### Querying file-level metadata kv-pairs

clp-json stores some metadata about each file that's compressed into an archive (e.g., the file's
name). To filter for log events that correspond to some kv-pair in this metadata, you can prefix
the key with `$`:

```
$key: value
```

For example, to query for log events that were compressed from a file whose name was `test.jsonl`:

```
$_filename: "test.jsonl"
```

To query for a key named `$key` within the default namespace, the `$` character can be escaped as
follows:

```
\@key: value
```
````

I think for a normal user, this is as much information as they would need. What do you think?

Author: That's fair. I guess I can leave info about how the metadata search actually works for developer docs.

\@key: value
```

### Querying file-level metadata kv-pairs

> clp-json stores some metadata about each file that's compressed into an archive (e.g., the file's
> name). To filter for log events that correspond to some kv-pair in this metadata, you can prefix
> the key with `$`:

Reviewer 2: ```suggestion
clp-json stores some metadata about each file that is compressed into an archive (e.g., the file's
name). To filter for log events that correspond to some kv-pair in this metadata, you can prefix the
key with `$`:
```

```
$key: value
```

For example, to query for log events that were compressed from a file whose name was `test.jsonl`:

```
$_filename: "test.jsonl"
```

> To query for a key named `$key` within the default namespace, the `$` character can be escaped as
> follows:

Reviewer 2: ```suggestion
To query for a key named `$key` in an event, the `$` character can be escaped as follows:
```
